clouduniqueid/clouds/bitbucket/__init__.py

```diff
@@ -0,0 +1,51 @@
+import logging
+import re
+from typing import Dict
+
+from .uniqueid import unique_id_patterns
+
+logger = logging.getLogger(__name__)
+
+
+def check_missing_data_keys(dataKeys: list, uniqueIdFormat: str):
+    uniqueIdKeys = re.findall(r"{(.*?)}", uniqueIdFormat)
+    missingKeys = [key for key in uniqueIdKeys if key.lower() not in dataKeys]
+    return missingKeys
+
+
+class BitbucketUniqueId:
+    def get_unique_id(
+        self,
+        resourceType: str,
+        data: Dict,
+        service: str = "bitbucket",
+    ) -> str:
+        data = {k.lower().replace("_", ""): v for k, v in data.items()}
+        data_keys = list(data.keys())
+
+        if service not in unique_id_patterns:
+            logger.error(f"Service {service} unknown")
+            return ""
+
+        if resourceType not in unique_id_patterns[service]:
+            logger.error(
+                f"Service {service} resource type {resourceType} not supported",
+            )
+            return ""
+
+        unique_id_format = unique_id_patterns[service][resourceType]
+        missing_keys = check_missing_data_keys(data_keys, unique_id_format)
+
+        if missing_keys:
+            error_msg = ", ".join(missing_keys)
+            logger.error(f"Bitbucket {error_msg} keys required")
+            return ""
+
+        return eval(f"f'{unique_id_format}'".format(**data)).replace(" ", "")
```

Comment:
<b>Vulnerability:</b> Detected the use of eval(). eval() can be dangerous if used to evaluate dynamic content. If this content can be input from outside the program, this may be a code injection vulnerability. Ensure evaluated content is not definable by external sources.<br/>
                            <b>Lines</b>: <b>Lines</b>: 44 to 44<br/>
                            <b>OWASP:</b> A03:2021 - Injection<br/>
                            <b><a href="https://console.cloudanix.com/code?t=vulnerabilities" target='_blank'>Click here for more details</a></b>